src/wamr/pthread.cpp

```diff
@@ -1,27 +1,66 @@
 #include <wamr/native.h>
 #include <wasm_export.h>
 
+#include <faabric/util/logging.h>
+
 namespace wasm {
-// ------------------------------------------
-// NOTE: these functions are just here to stub out bits that are missing
-// from the existing WAMR pthread implementation. This can be found here:
-// https://github.com/bytecodealliance/wasm-micro-runtime/tree/main/core/iwasm/libraries/lib-pthread
-// ------------------------------------------
+static int32_t pthread_mutex_init_wrapper(wasm_exec_env_t exec_env,
+                                          int32_t a,
+                                          int32_t b)
+{
+    faabric::util::getLogger()->debug("S - pthread_mutex_init {} {}", a, b);
```

Comment:
This is the WAMR implementation of these functions, and we don't support multi-threading with WAMR yet, therefore it's not possible to execute multi-threaded applications that will invoke these. However, we can't throw an exception as they will always be called by malloc/ free now that we use thread-safe dlmalloc. I guess we could add the implementations now, but there is a lot more work to be done to make the WAMR implementation support multi-threading. I think a comment here to explain this might be better.